scipy/optimize/optimize.py

```diff
@@ -974,14 +992,40 @@ def check_grad(func, grad, x0, *args, **kwargs):
     >>> from scipy.optimize import check_grad
     >>> check_grad(func, grad, [1.5, -1.5])
     2.9802322387695312e-08
+    >>> rng = np.random.default_rng()
+    >>> check_grad(func, grad, [1.5, -1.5], 
+    ...             random_projection=True, seed=rng)
+    2.9802322387695312e-08
 
     """
+    x0 = np.asarray(x0)
     step = kwargs.pop('epsilon', _epsilon)
+    random_projection = kwargs.pop('random_projection', False)
+    if random_projection:
+        random_state = check_random_state(kwargs.pop('seed', None))
     if kwargs:
-        raise ValueError("Unknown keyword arguments: %r" %
+        raise ValueError("Unexpected keyword arguments: %r" %
                          (list(kwargs.keys()),))
-    return sqrt(sum((grad(x0, *args) -
-                     approx_fprime(x0, func, step, *args))**2))
+    
+    def g(w, *args):
+        func, x0, v = args[0:3]
+        return func(x0 + w*v, *args[3:])
+
+    if random_projection:
+        v = rng_integers(random_state, 0, 2, size=x0.shape)
+        v[v == 0] = -1
+        _args = (func, x0, v) + args
+        _func = g
+        vars = np.zeros((1,))
```

Comment:
I'm a little puzzled. Why is the gradient evaluated at w=1 rather than w=0, which corresponds to the user provided location?

If the tests pass with w=1, then they are probably not strict enough

Reply:
The gradient is evaluated at w=0. 
```python
>>> import numpy as np
>>> np.zeros((1,))
array([0.])
```
Here instead of `x` we have a only 1 dimensional `w` and the existing code in other functions will work without change.

Reply:
Oops, I misread this. Please ignore!